src/app/shared/util.ts

```diff
@@ -0,0 +1,62 @@
+import { Injectable } from '@angular/core';
+
+@Injectable()
+export class FileService {
+
+    constructor() {}
+
+    isCSVFile(file) {
+        return file.name.endsWith('.csv');
+    }
+
+    getHeaderArray(csvRecordsArr, tokenDelimeter) {
+        const headers = csvRecordsArr[0].split(tokenDelimeter);
+        const headerArray = [];
+        for (let j = 0; j < headers.length; j++) {
+            headerArray.push(headers[j]);
+        }
+        return headerArray;
+    }
+
+    validateHeaders(origHeaders, fileHeaders) {
+        if (origHeaders.length !== fileHeaders.length) {
+            return false;
+        }
+
+        let fileHeaderMatchFlag = true;
```

Comment:
Dont we need else here? Else the subsequent stms will be always executed rt?

Reply:
If the headers don't match it will return false and the subsequent statement won't be executed. The control will go to the function that has called validateHeader method.